src/DistributedComputations/distributed_transpose.jl

```diff
@@ -0,0 +1,113 @@
+using Oceananigans.Grids: architecture
+using Oceananigans.Architectures: on_architecture
+using KernelAbstractions: @index, @kernel
+using MPI: VBuffer, Alltoallv!
+
+# Transpose directions are assumed to work only in the following configuration
+# z -> y -> x -> y -> z
+# where z stands for z-complete data, y for y-complete data, and x for x-complete data
```

Comment:
After reading this PR I understand the terminology introduced here to be

"x-complete" means that a field has been transposed so that the x-direction is entirely local (implying that in its "native" configuration, the x-direction is not local).

is that right?

Reply:
yep, that is the case

Reply:
Can you then write something like

> where "z" indicates a data configuration in which the "z" direction is entirely local (this is the "native" configuration in which all other computations are done). We refer to this as "z-complete". Similarly "x" and "x-complete" refer to the _transposed_ data configuration in which the x-direction is entirely local.

Writing this I wonder if a better phrase is "x-local" or something.

Reply:
x-local is nice